src/plugins/intel_gpu/src/kernel_selector/kernels/convolution/convolution_kernel_b_fs_yx_fsv16_1x1.cpp

```diff
@@ -22,22 +22,28 @@ ConvolutionKernel_b_fs_yx_fsv16_1x1::ConvolutionKernel_b_fs_yx_fsv16_1x1() : Con
 
 ConvolutionKernel_b_fs_yx_fsv16_1x1::AutoTuneOption ConvolutionKernel_b_fs_yx_fsv16_1x1::GetAutoTuneOptions(const Params& params,
                                                                                                             int /*autoTuneIndex*/) const {
-    const convolution_params& cp = static_cast<const convolution_params&>(params);
-
-    auto x = cp.outputs[0].X().v;
-    auto y = cp.outputs[0].Y().v;
-    auto f = cp.outputs[0].Feature().v;
-
-    if (x == 1 && y == 1) {
-        return { 1, EXE_MODE_DEFAULT };
-    } else if (x * f <= 256) {
-        if (x < 8 || x * f <= 128)
-            return { 2, EXE_MODE_DEFAULT };
-        else
+    if (!params.is_shape_agnostic) {
+        const convolution_params& cp = static_cast<const convolution_params&>(params);
+
+        auto x = cp.outputs[0].X().v;
+        auto y = cp.outputs[0].Y().v;
+        auto f = cp.outputs[0].Feature().v;
+
+        if (x == 1 && y == 1) {
+            return { 1, EXE_MODE_DEFAULT };
+        } else if (x * f <= 256) {
+            if (x < 8 || x * f <= 128)
+                return { 2, EXE_MODE_DEFAULT };
+            else
+                return { 4, EXE_MODE_DEFAULT };
+        } else if (x * f <= 1536) {
             return { 4, EXE_MODE_DEFAULT };
-    } else if (x * f <= 1536) {
-        return { 4, EXE_MODE_DEFAULT };
+        } else {
+            return { 8, EXE_MODE_DEFAULT };
+        }
     } else {
+        // In shape agnostic kernel, the output shape cannot be specified at build time,
+        // So we set blockWidth to 8, which is the most commonly used.
```

Comment:
How about using multi-kernel approach as in gemm_tiled_opt? I think that current assumption is not safe.
Please find the attached for my suggested changes.
[diff.patch](https://github.com/user-attachments/files/15959880/diff.patch)

Reply:
Good idea, applied.